### PR TITLE
Hw2 dev

### PR DESCRIPTION
**Pull request by ZhenruWang**

### CODE REVIEW

**Comment by lindseysbrown** [comment]
Problem 1: All screenshots correct.  15/15

**Comment by cj-xin** [comment]
### Problem 2

3/3 pt: Plot is correct
2/2 pt: File read in correctly (recommend using `with(...)` block
1/1 pt: Raw data processed correctly
1/1 pt: Correct counting of occurrences
1/1 pt: Results printed as specified
1/2 pt: Plotting should be handled outside of function

Comment: Comment code as appropriate

File: homeworks/HW2-final/P6.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+def timer(f):
+    def inner(*args):
+        t0 = time.time()
+        output = f(*args)
+        elapsed = time.time() - t0
+        print("Time Elapsed", elapsed)
+        return output
+    return inner
+
+def my_ave(areas):
```
Comment (bhavenp) [inline homeworks/HW2-final/P6.py:14]: -2 points: `my_ave` and `np_ave` should be decorated with the `timer` function.

File: homeworks/HW2-final/P6.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+def np_ave (areas):
+    return np.mean(areas)
+
+file = open('circles.txt')
+list_str= file.read().splitlines()
+radius = [ float(x) for x in list_str ]
+
+areas = [0.0]*len(radius)
+areas_np = np.zeros(len(radius))
+for i in range(len(radius)):
```
Comment (bhavenp) [inline homeworks/HW2-final/P6.py:32]: Use `enumerate` when you are looping over a list and want both the index and item.

**Comment by feiyu-chen96** [comment]
Problem 7
The decorator (3/5) -2 The decorated function should still return the output of its un-decorated version
The decorated functions (4/4) Good job!
The tests (6/6) Good job!

**Comment by adityakaran** [comment]
P4: 8/10 Need to test the invalid case. Test should be clearer to the end user (what's the input/output?)

**Pull request merged**